crates/forge/bin/cmd/test/mod.rs

```diff
@@ -814,6 +828,50 @@ fn persist_run_failures(config: &Config, outcome: &TestOutcome) {
     }
 }
 
+/// Generate test report in JUnit XML report format.
+fn junit_xml_report(results: &BTreeMap<String, SuiteResult>, verbosity: u8) -> Report {
+    let mut total_duration = Duration::default();
+    let mut junit_report = Report::new("Test run");
+    junit_report.set_timestamp(Utc::now());
+    for (suite_name, suite_result) in results {
+        let mut test_suite = TestSuite::new(suite_name);
+        total_duration += suite_result.duration;
+        test_suite.set_time(suite_result.duration);
+        test_suite.set_system_out(suite_result.summary());
+        for (test_name, test_result) in &suite_result.test_results {
+            let test_status = if test_result.status.is_failure() {
+                let mut status = TestCaseStatus::non_success(NonSuccessKind::Failure);
+                status.set_message(test_result.reason.clone().unwrap_or_default());
+                status
+            } else if test_result.status.is_skipped() {
+                TestCaseStatus::skipped()
+            } else {
+                TestCaseStatus::success()
+            };
+
+            let mut test_case = TestCase::new(test_name, test_status);
+            test_case.set_time(test_result.duration);
+
+            let mut sys_out = String::new();
+            let result_report = test_result.kind.report();
+            write!(sys_out, "{test_result} {test_name} {result_report}").unwrap();
+            if verbosity >= 2 && !test_result.logs.is_empty() {
+                write!(sys_out, "\\nLogs:\\n").unwrap();
+                let console_logs = decode_console_logs(&test_result.logs);
+                for log in console_logs {
+                    write!(sys_out, "  {log}\\n").unwrap();
```

Comment:
are the new lines not escaped by quick-junit?

Reply:
they are, but the indent is not preserved for closing tag and result in smth like

```xml
    <testsuite name="test/Reverter.t.sol:NonceTest" tests="1" disabled="0" errors="0" failures="1" time="0.066">
        <testcase name="test_junit5(uint256)" time="0.064">
            <failure message="EvmError: Revert"/>
            <system-out>
[FAIL. Reason: EvmError: Revert; counterexample: calldata= args=[100]] test_junit5(uint256) (runs: 185, μ: 3478, ~: 3478)\nLogs:\n  junit3
</system-out>
        </testcase>
        <system-out>Suite result: FAILED. 0 passed; 1 failed; 0 skipped; finished in 65.80ms (64.39ms CPU time)</system-out>
    </testsuite>
```

so thought would be nicer this way... Can remove and leave it like this, wdyt?

Reply:
huh weird, this is fine then